### ISSUE TITLE
How to make use of GRL for speaker	

### ISSUE THREAD

**Issue by gtonkov** [PlayVoice/so-vits-svc-5.0#19]
Hi,

How can I use the SpeakerClassifier in vits/modules_grl.py?

It was added with this [commit](https://github.com/PlayVoice/so-vits-svc-5.0/commit/4f57275c373e5d46ee8be96a2bf0d903e5ce1416) but I cannot see this to be used anywhere during training. 

Thanks!

**Comment by MaxMax2016** [comment]
https://github.com/PlayVoice/so-vits-svc-5.0/blob/main/vits/modules_grl.py
it will be used to finetune model, after base model trained. I still train base model, now.

this code will be added for the finetune
# -- Speaker Classifier
# speaker_posterior = self.speaker_classifier(prosody_vector.squeeze(1))
# LOSS
# self.nll_loss = nn.NLLLoss()
# adv_loss = self.nll_loss(speaker_posteriors, speaker_targets[序号])

more info for GRL can be finded here:https://github.com/ubisoft/ubisoft-laforge-daft-exprt/blob/master/src/daft_exprt/model.py

**Comment by MaxMax2016** [comment]
At the finetune stage, PosteriorEncoder should be frozen, if you want to try GRL now.

**Comment by gtonkov** [comment]
great, thanks :)
